mlir/tools/quantum-opt/CMakeLists.txt

```diff
@@ -10,6 +10,23 @@ set(LIBS
     quantum-transforms
     MLIRGradient
     gradient-transforms
+
+    MhloRegisterDialects
+    StablehloRegister
+
+    # Taken from mlir-hlo/bindings/c/CMakeLists.txt:
+    # For AllMhLoPasses:
+    ChloPasses
+    MhloPasses
+    MhloToLhloConversion
+    MhloToArithmeticConversion
+    MhloToMemrefConversion
+    MhloToStandard
+    MhloToThloConversion
+    MhloToLinalg
+    MhloToStablehlo
+    MhloShapeOpsToStandard
+    StablehloToMhlo
```

Comment:
As above.

Hmm.. if this is a common practice, then I suggest to use a CMake set variable to define them all once at the top CMakeLists.txt of `./mlir` and reuse them in these CMake files. It make the code maintenance easier and we can keep such dependencies in one place